rcl_action/include/rcl_action/goal_state_machine.h

```diff
@@ -83,23 +80,38 @@ _set_canceled_event_handler(rcl_action_goal_state_t state, rcl_action_goal_event
 }
 
 // Transition map
-rcl_action_goal_event_handler
-  _goal_state_transition_map[GOAL_STATE_NUM_STATES][GOAL_EVENT_NUM_EVENTS] = {
-  [GOAL_STATE_ACCEPTED] = {
-    [GOAL_EVENT_EXECUTE] = _execute_event_handler,
-    [GOAL_EVENT_CANCEL] = _cancel_event_handler,
-  },
-  [GOAL_STATE_EXECUTING] = {
-    [GOAL_EVENT_CANCEL] = _cancel_event_handler,
-    [GOAL_EVENT_SET_SUCCEEDED] = _set_succeeded_event_handler,
-    [GOAL_EVENT_SET_ABORTED] = _set_aborted_event_handler,
-  },
-  [GOAL_STATE_CANCELING] = {
-    [GOAL_EVENT_SET_SUCCEEDED] = _set_succeeded_event_handler,
-    [GOAL_EVENT_SET_ABORTED] = _set_aborted_event_handler,
-    [GOAL_EVENT_SET_CANCELED] = _set_canceled_event_handler,
-  },
-};
+static rcl_action_goal_event_handler
+  _goal_state_transition_map[GOAL_STATE_NUM_STATES][GOAL_EVENT_NUM_EVENTS];
+
+static inline void
+rcl_action_goal_transition_map_init()
```

Comment:
@jacobperron Why are we using static linkage for these? Why not defining them on goal_state_machine.c ?

Reply:
Yeah, agreed; for a large function like this, making it static inline is bloating the downstream users for not a lot of reason.  This is probably better as a function inside a C file.

Reply:
In retrospect, I think all of these static functions and the transition map can be moved to a C file.

Reply:
+1, you should move as much out of the public interface as is possible/reasonable.